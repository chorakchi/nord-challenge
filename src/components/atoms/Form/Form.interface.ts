import { FormHTMLAttributes } from "react";

export interface FormProps extends FormHTMLAttributes<HTMLFormElement> {}
