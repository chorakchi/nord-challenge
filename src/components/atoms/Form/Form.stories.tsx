import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Form } from "./Form";

export default {
  title: "Atoms/Form",
  component: Form,
  parameters: {
    docs: {
      description: {
        component:
          `This component doesn't have visual style and it's a wrapper on form element.<br/>` +
          `<br/> <br/> <h5> Form props:</h5>` +
          " Standard form attributes are supported"
      },
    },
    backgrounds: {
      default: "light",
    },
  },
} as ComponentMeta<typeof Form>;

const Template: ComponentStory<typeof Form> = (args) => (
  <Form {...args} />
);

export const Form_ = Template.bind({});
Form_.args = {};
