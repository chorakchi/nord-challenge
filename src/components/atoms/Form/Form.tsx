import { FormProps } from "./Form.interface";
export const Form = ({ children, ...props }: FormProps) => {
  return (
    <form {...props} className="mt-8 space-y-6" action="#" method="POST">
      <input type="hidden" name="remember" value="true" />
      {children}
    </form>
  );
};
