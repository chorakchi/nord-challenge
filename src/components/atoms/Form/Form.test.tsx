import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Form } from "./Form";

describe("Form", () => {
  it("should render the Form", () => {
    const subject = shallow(<Form />);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Form />).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
