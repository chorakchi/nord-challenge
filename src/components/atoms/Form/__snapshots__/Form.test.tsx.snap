// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly and match to the Snapshot 1`] = `
<form
  action="#"
  className="mt-8 space-y-6"
  method="POST"
>
  <input
    name="remember"
    type="hidden"
    value="true"
  />
</form>
`;
